### ISSUE TITLE
Error with the synth selection dialog

### ISSUE THREAD

**Issue by CyrilleB79** [nvaccess/nvda#13042]
Sometimes a Wx error occurs when closing the synthesizer selection dialog. A reliable way to reproduce the error has been found thanks to @paulber19.

### Steps to reproduce:

1. Start NVDA (with no add-ons); the default TTS should be OneCore.
2. Open the Speech settings panel (NVDA+ctrl+V)
3. Tab to the "Change" button
4. Select "Microsoft Speech API version 5" and validate the dialog
5. Tab to the "OK" button and press it to validate the Speech settings panel.
6. Execute again steps 2. to 4., but this time, select OneCore synthesizer instead of SAPI5.

### Actual behavior:

An error occurs in the log:
`RuntimeError: wrapped C/C++ object of type BoxSizer has been deleted`

Here is the full log:
[nvda[ErrorChangeSynth].log](https://github.com/nvaccess/nvda/files/7500229/nvda.ErrorChangeSynth.log)

### Expected behavior:
No error.

#### NVDA installed/portable/running from source:
installed

#### NVDA version:
2021.3beta2
#### Windows version:
Windows 10 2004 (x64) build 19041.388

#### Name and version of other software in use when reproducing the issue:
N/A
#### Other information about your system:
N/A
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
I haven't personally tested this exact STR on other NVDA versions. But I recall having a similar issue in older stable NVDA versions.
Also @paulber19 has indicated to have performed the tests with NVDA 2019.3 and 2021.2 and has confirmed that the issue was already present in this version (cf [this thread](https://groups.io/g/script-NVDA/topic/86901549#591) on a French mailing list).
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Not tested but probably not related.

**Comment by CyrilleB79** [comment]
Cc @seanbudd since you have worked on similar GUI issues.

**Comment by CyrilleB79** [comment]
I have tested last master after the merge of #16079. Unfortunately, the issue is still there.

@lukaszgo1, in case it is related to your recent work, maybe you want to have a look?

**Comment by Adriani90** [comment]
This is also reproducible on 64 bit NVDA right?

**Comment by paulber19** [comment]
Yes.
